README.md

```diff
@@ -112,3 +113,25 @@ And in json format
   }
 }
 ```
+
+# Guidelines for `packspec` implementers
+
+The minimum supported feature set to be considered `packspec`-compatible is:
+
+## Managing package versions
+- Must be able to fetch and parse `packspec.json` files
+- Must be able to use `git` to retrieve and manipulate package sources
+- Must be able to fetch tagged commits for specified package versions
+- Must be able to check for updated `packspec.json` files
+
+## Managing Neovim dependencies
+
+- Must be able to check the current version of `Neovim` and warn on incompatibility
+- Must be able to retrieve and manage the specified versions of dependencies transitively, starting from user-specified packages
+- Must either be able to solve for compatible versions of dependency packages across all dependency relationships, or warn users if using a potentially inconsistent version resolution strategy (e.g. picking the first specified version of a dependency).
+- Must be able to remove dependencies when they are no longer required (transitively) by any user-specified packages
```

Comment:
Pip doesn't even do this so I'm not sure we should require it.

Reply:
Which one?

Reply:
"Must be able to remove dependencies..."

Reply:
From `packer`'s perspective, this isn't very hard to do. And personally, I don't think `pip` is a good model for package management - it has a lot of pain points and missing features (this among them, in my opinion).

Reply:
Hmm ok. Well nothing is stopping implementations from doing this, just not sure it needs to be in the spec.

Reply:
"Remove" as in garbage collection? I don't think it should be a requirement.

Reply:
Maybe we just strongly recommend it?